lib/ecto/schema.ex

```diff
@@ -1174,17 +1174,25 @@ defmodule Ecto.Schema do
   @doc false
   def __belongs_to__(mod, name, queryable, opts) do
     check_options!(opts, @valid_belongs_to_options, "belongs_to/3")
+    opts = set_foreign_key!(mod, name, opts)
 
+    struct =
+      association(mod, :one, name, Ecto.Association.BelongsTo, [queryable: queryable] ++ opts)
+    Module.put_attribute(mod, :changeset_fields, {name, {:assoc, struct}})
+  end
+
+  defp set_foreign_key!(mod, name, opts) do
```

Comment:
@josevalim -- no need to break it up. Thought it might be bit more readable. You prefer as it was? :)